redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractBoxTest.kt

```diff
@@ -364,6 +364,26 @@ abstract class AbstractBoxTest<T : Any> {
     verifySnapshot(widget.value, "Empty")
   }
 
+  /** The view shouldn't crash if its displayed after being detached. */
+  @Test
+  fun testLayoutAfterDetach() {
+    val widget = box().apply {
+      width(Constraint.Wrap)
+      height(Constraint.Wrap)
+      horizontalAlignment(CrossAxisAlignment.Start)
+      verticalAlignment(CrossAxisAlignment.Start)
+      children.insert(
+        0,
+        text().apply {
```

Comment:
I think a good test to add is one with a bunch of children, all with different modifiers to make sure they don't clobber each other or get shared when we don't want them to. (specifically .height and .width ones)

Reply:
Great idea

Reply:
I’m gonna send a PR with this test as a fast-follow-up!

Reply:
https://github.com/cashapp/redwood/pull/2045